src/App.tsx

```diff
@@ -1,24 +1,49 @@
 import React from 'react';
 import './App.scss';
+import Clock from './Clock/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+type Props = {};
+
+interface State {
+  isClockVisible: boolean;
+  clockName: number;
+}
+
+class App extends React.Component<Props, State> {
+  state = {
+    isClockVisible: true,
+    clockName: Math.floor(Math.random() * 100),
+  };
+
+  showClock = () => this.setState({ isClockVisible: true });
+
+  hideClock = () => this.setState({ isClockVisible: false });
+
+  renameClock = () => this.setState({ clockName: Math.floor(Math.random() * 100) });
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {isClockVisible && <Clock clockName={clockName} />}
+
+        <div className="buttons">
+          <button className="buttons__btn" type="button" onClick={this.showClock}>
+            Show Clock
+          </button>
+          <button className="buttons__btn" type="button" onClick={this.hideClock}>
+            Hide Clock
+          </button>
+          <button className="buttons__btn" type="button" onClick={this.renameClock}>
+            Rename Clock
+          </button>
```

Comment:
I still can see and click Hide and Set random name buttons when the clock is hidden, even though they are having no effect when clock is hidden, so either hide them or make disabled based on state